turbomoleio/output/data.py

```diff
@@ -1484,3 +1500,94 @@ def from_parser(cls, parser):
             return None
 
         return cls(**data)
+
+
+class PeriodicityData(BaseData):
+    """
+    Information about periodicity
+    """
+
+    def __init__(self, periodicity=None, lattice_params=None,
+                 shortest_interatomic_distance=None,
+                 direct_space_vectors=None, reciprocal_space_vectors=None):
+        """
+        Args:
+            periodicity (int): Periodicity of the system (1, 2 or 3).
+            lattice_params: Lattice parameters of the system (in Angstroms).
+                For 1D systems, a single number: [a]
+                For 2D systems, three numbers: [a, b, gamma], i.e. the two lattice
+                    parameters and the angle between them.
+                For 3D systems, three numbers: [a, b, c, alpha, beta, gamma], i.e.
+                    three two lattice parameters and the angles between them.
+            shortest_interatomic_distance: Shortest interatomic distance in the
+                system (in Angstroms).
+            direct_space_vectors: Lattice vectors of the system (in Angstroms).
+                The first vector will always be aligned with the x cartesian direction.
+                For 2D and 3D, the second vector is always in the xy cartesian plane.
+            reciprocal_space_vectors: Reciprocal lattice vectors of the system (in Angstroms^-1).
+                The physics definition of the reciprocal lattice vectors is used here, i.e.
+                    b1 = 2*pi/V * (a2 x a3)
+                    b2 = 2*pi/V * (a3 x a1)
+                    b3 = 2*pi/V * (a1 x a2)
+                The crystallographer's definition is easily recovered as:
+                    b'1 = b1 / (2*pi)
+                    b'2 = b2 / (2*pi)
+                    b'3 = b3 / (2*pi)
+
+        Returns:
+            PeriodicityData.
+
+        """
+        self.periodicity = periodicity
+        self.lattice_params = lattice_params
+        self.shortest_interatomic_distance = shortest_interatomic_distance
+        self.direct_space_vectors = direct_space_vectors
+        self.reciprocal_space_vectors = reciprocal_space_vectors
+
+    @classmethod
+    def from_parser(cls, parser):
+        """
+        Generates an instance of PeriodicityData from a parser based on the stdout
+        of a Turbomole executable. Returns None if no data could be parsed.
+
+        Args:
+            parser (Parser): the parser to be used to extract the data.
+
+        Returns:
+            PeriodicityData.
+        """
+        data = parser.periodicity_data
+        if not data:
+            return None
+
+        periodicity = data["periodicity"]
+        if periodicity == 1:
+            # a
+            lattice_params = [bohr_to_ang * data["tm_lattice_params"][0]]
+        elif periodicity == 2:
+            # a, b, gamma
```

Comment:
maybe  here it should also store the data in a way that is consistent with the `$lattice` format? Otherwise we can have inconsistent data with respect to the coordinates.

Reply:
Seems like when you use $lattice with e.g.
$lattice 
  5.0 -5.0
  5.0 5.0
which is not "compatible" with $cell, at least it is not exactly equivalent, then in the output, it gives the a, b, gamma values ("correct ones").

Did I understand correctly your comment ?

Reply:
if the information about `$lattice` is not accessible from the output there is not much that we can do. We can only save this. But how is it dealt with in the text? Because if you just give the parameters and the cartesian coordinates one will not know how to relate them. Anyway, if they decided not to print the whole information this is rather a problem in TM.

I also have another doubt: should the data be stored always as a list of six elements (with some `None`s in case of periodicity 1 or 2), instead of a list with variable length as it is now?

Reply:
Well it is actually there in the direct_space_vectors. If you use $cell to define the lattice, then you get the tm_lattice_parameters corresponding to the cell + the "TM-conventioned" direct lattice vectors. If you use $lattice, then the direct lattice vectors are this exact $lattice and the tm_lattice_parameters is "deduced" from this (but I think internally it is using the lattice as defined in $lattice of course). So to summarize, I think the relevant information is in the direct_lattice_vectors, because it holds the correct information whichever $cell of $lattice was used in the control file.

As for the list of 6 elements, I don't know. Why do you think it would be better to do it this way ?

Reply:
ah sorry, I missed that the `direct_space_vectors` contained that information. It is fine then.

Concerning the 6 elements, I thought it would be a bit more easy to access having a uniform format across the different periodicities. e.g. no risk of an index out of range error and you know that the 3rd element is always a length (or None), and not an angle or a length or non existing. But both solutions should be fine for storing purposes.

Reply:
So for 1d it would be: [a, None, None, None, None, None]
for 2d: [a, b, None, None, None, gamma] (and not [a, b, gamma, None, None, None] right ?)
and 3d of course: [a, b, c, alpha, beta, gamma]
I have no real preference. As you say its just for storing so we could just keep it as it is. Just tell me :)